sklearn/tree/_criterion.pyx

```diff
@@ -292,11 +287,8 @@ cdef class ClassificationCriterion(Criterion):
 
         Parameters
         ----------
-        y : array-like, dtype=DOUBLE_t
+        y : array-like, dtype=const DOUBLE_t[:, ::1]
```

Comment:
The dtype is still `DOUBLE_t` here. How about a more descriptive:

```
y : C-contiguous 2D const memory view, dtype=DOUBLE_t
```

Note that other docstrings need to be updated e.g. the one of `Criterion.init`.

Reply:
I think the original version is fine. We put the dtype in docstrings because python isn't statically typed, but Cyhon can be, so in that case providing the `type` description both in the function definition and the docstrings is redundant and can result in things getting out of sync (particularly if they are worded differently).

Reply:
ok, then keeping the original. These are not rendered anyway, and I guess it's only useful for the developers, who would see the function/method's signature.